packages/eui/.storybook/addons/code-snippet/decorators/utils.ts

```diff
@@ -0,0 +1,218 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+/* DISCLAIMER: Parts of this file were copied from Storybook jsxDecorator and adjusted for more specific needs.
+https://github.com/storybookjs/storybook/blob/2bff7a1c156bbd42ab381f84b8a55a07694e7e53/code/renderers/react/src/docs/jsxDecorator.tsx#L224 */
+
+import { ReactElement, FunctionComponent, ComponentType } from 'react';
+import * as prettier from 'prettier';
+import tsParser from 'prettier/parser-typescript';
+import { StoryContext } from '@storybook/react';
+
+// @ts-ignore - config import
+import basePrettierConfig from '../../../../.prettierrc';
+
+export const toPascalCase = (str: string) =>
+  str.charAt(0).toUpperCase() + str.slice(1);
+
+/* Helpers for React specific checks */
+export const isMemo = (component: any) =>
+  component.$$typeof === Symbol.for('react.memo');
+export const isForwardRef = (component: any) =>
+  component.$$typeof === Symbol.for('react.forward_ref');
+export const isFragment = (component: any) =>
+  component.type?.toString().includes('fragment') ||
+  component.$$typeof?.toString().includes('fragment');
+
+/* Helpers */
+// returns the displayName and handles typing as
+// otherwise `type` would not be typed
+export const getElementDisplayName = (
+  node: ReactElement<any>
+): string | undefined => {
+  let displayName;
+
+  if (typeof node.type === 'function' || typeof node.type === 'object') {
+    displayName =
+      (node.type as FunctionComponent).displayName ??
+      (node.type as FunctionComponent).name ??
+      undefined;
```

Comment:
not sure how much you care about this, both feels pretty readable to me but one is just a touch more DRY
```suggestion
    const component = node.type as FunctionComponent;
    displayName = component.displayName ?? component.name ?? undefined;
```